ee/zentral/contrib/intune/api_client.py

```diff
@@ -0,0 +1,238 @@
+        return ms_tree
+
+    # helper methods
+
+    def get_business_unit_d(self):
+        return self.business_unit.serialize()
+
+    def get_source_d(self):
+        return {
+            "module": "zentral.contrib.intune",
+            "name": "Intune",
+            "config": {
+                "tenant_id": self.tenant_id
+            }
+        }
+
+    def add_ms_tree_extra_facts(self, ms_tree, device):
+        extra_facts = {}
+        compliance_status = device.compliance_state
+        if isinstance(compliance_status, str):
+            extra_facts["compliance_status"] = compliance_status
+        partner_reported_threat_state = self.add_ms_tree_extra_facts_partner_reported_threat_state(device)
+        if isinstance(partner_reported_threat_state, str):
+            extra_facts["partner_reported_threat_state"] = partner_reported_threat_state
+        extra_facts["azure_ad_device_id"] = device.azure_a_d_device_id
+        if extra_facts:
+            ms_tree["extra_facts"] = extra_facts
+
+    def add_ms_tree_extra_facts_partner_reported_threat_state(self, device):
+        match device.partner_reported_threat_state:
+            case 0:
+                return "Unknown"
+            case 1:
+                return "Activated"
+            case 2:
+                return "Deactivated"
+            case 3:
+                return "Secured"
+            case 4:
+                return "Low Severity"
+            case 5:
+                return "Medium Severity"
+            case 6:
+                return "High Severity"
+            case 7:
+                return "Unresponsive"
+            case 8:
+                return "Compromised"
+            case 9:
+                return "Misconfigured"
+        return None
+
+    def add_ms_tree_os_version(self, ms_tree, device):
+
+        def apple_version_tuple_to_parse(device):
+            try:
+                os_version_tuple = tuple(int(s) for s in device.os_version.split()[0].split('.'))
+                os_version = dict(zip(('major', 'minor', 'patch'), os_version_tuple))
+            except (TypeError, ValueError):
+                logger.warning("Device %s: could not parse OS version", device.id)
+                return
+            try:
+                os_build = device.os_version.split()[1].translate(str.maketrans("", "", "()"))
+                if os_build:
+                    os_version["build"] = os_build
+            except (TypeError, ValueError):
+                logger.warning("Device %s: could not parse build version", device.id)
+            return os_version_tuple, os_version
+
+        manufacturer = device.manufacturer
+        if manufacturer == "Apple":
+            os_name = "iOS"
+            os_version_tuple, os_version = apple_version_tuple_to_parse(device)
+            if "ipad" in device.model.lower() and os_version_tuple >= (13, 1):
+                os_name = "iPadOS"
+            os_version["name"] = os_name
+        elif manufacturer == "AppleOsX":
```

Comment:
This AppleOsX thing is a WSOne specific value. I do not think this value exist in Intune.

Reply:
I'll remove it. Thanks.